### PR TITLE
feat(rust): replacing ffi_support

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

File: implementations/rust/ffi/include/kex.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
+//
+// Created by Oleksandr Deundiak on 10.09.2020.
```
Comment (mikelodder7) [inline implementations/rust/ffi/include/kex.h:2]:
```suggestion
// Created by Ockam Developers
```

File: implementations/rust/ffi/src/kex.rs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+/// Create a new kex initiator and return it
+#[no_mangle]
+pub extern "C" fn ockam_kex_xx_initiator(
+    context: &mut FfiKexFatPointer,
+    vault: FfiVaultFatPointer,
+    identity_key: u64,
+) -> FfiOckamError {
+    let res = || {
```
Comment (mikelodder7) [inline implementations/rust/ffi/src/kex.rs:54]: I don't understand this. Why are you creating an anonymous function then immediately calling it?
What's the purpose of the anonymous function?

Comment (SanjoDeundiak) [reply implementations/rust/ffi/src/kex.rs:54]: The purpose is to allow using ? operator inside

**Review by mikelodder7 — APPROVED** [review]

File: implementations/rs/common/src/error.rs [hunk 3]
```diff
@@ -19,7 +19,7 @@ impl OckamError {
         self.code
     }
     /// Error domain
-    pub fn domain(&self) -> &str {
+    pub fn domain(&self) -> &'static str {
```
Comment (mikelodder7) [inline implementations/rs/common/src/error.rs:22]: good change

**Review by mikelodder7 — APPROVED** [review]

**Pull request merged**